### PR TITLE
PN-8575: [FE] party switch refactoring for PA

### PR DESCRIPTION
**Pull request by SarahDonvito**
## Short description
Fixed issues and refactoring for party switch

## List of changes proposed in this pull request
- fixed change request of PR #1020
- managed errors in get institutions and get products in Header
- enhanced Header.tsx
- updated tests

## How to test
- Login in PA, switch between different products or different insitutions

### CODE REVIEW

File: packages/pn-commons/src/components/Header/__test__/Header.test.tsx [hunk 1]
```diff
@@ -100,30 +86,29 @@ describe('Header Component', () => {
     const productDropdown = await waitFor(() => screen.queryByRole('presentation'));
     expect(productDropdown).toBeInTheDocument();
     const products = productDropdown!.querySelectorAll('li');
-    fireEvent.click(products[1]);
+    await waitFor(() => fireEvent.click(products[1]));
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/Header/__test__/Header.test.tsx:89]: is the waitFor useful?

File: packages/pn-commons/src/components/Header/__test__/Header.test.tsx [hunk 2]
```diff
@@ -137,9 +122,9 @@ describe('Header Component', () => {
     const selfcareProductIndex = productsWithSelfcare.findIndex(
       (product) => product.id === 'selfcare'
     );
-    fireEvent.click(products[selfcareProductIndex]);
+    await waitFor(() => fireEvent.click(products[selfcareProductIndex]));
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/Header/__test__/Header.test.tsx:125]: is the waitFor useful?

File: packages/pn-commons/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -58,65 +53,42 @@ const Header = ({
...
-    window.location.assign(url);
-    trackEvent(url);
   };
 
-  const handlePartySelection = (party: PartyEntity) => {
-    window.location.assign(
-      `${selfcareBaseUrl}/token-exchange?institutionId=${party.id}&productId=${selfcareSendProdId}`
-    );
+  const handlePartySelection = (party: PartyEntityWithUrl) => {
+    window.location.assign(party.entityUrl);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/Header/Header.tsx:84]: add the check that party.entityUrl is defined and add the test case.
So leave a mocked entity with empty url. Add test case where we click on entity with empty url (no assign called) and click on entity with filled url (assign called)

File: packages/pn-commons/src/models/Institutions.ts [hunk 4]
```diff
@@ -51,3 +53,7 @@ interface RootParent {
   description: string;
   id: string;
 }
+
+export interface PartyEntityWithUrl extends PartyEntity {
+  entityUrl: string;
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/models/Institutions.ts:58]: add comment // only fe

File: packages/pn-pa-webapp/src/__mocks__/User.mock.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import { PartyEntity, ProductEntity } from '@pagopa/mui-italia';
 
+import { getConfiguration } from '../services/configuration.service';
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/__mocks__/User.mock.ts:3]: is this import useful?

File: packages/pn-commons/src/models/index.ts [hunk 6]
```diff
@@ -156,4 +158,7 @@ export type {
   PaymentsData,
   ExtRegistriesPaymentDetails,
   NotificationDetailTimelineDetails,
+  Institution,
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/models/index.ts:161]: Check that all these models are used outside the pn-commons

File: packages/pn-pa-webapp/src/api/external-registries/__test__/External-registries.api.test.ts [hunk 7]
```diff
@@ -38,9 +47,14 @@ describe('External registries api tests', () => {
   });
 
   it('getInstitutionProducts', async () => {
-    mock.onGet(GET_INSTITUTION_PRODUCTS('1')).reply(200, productsDTO);
+    const institutionId = '1';
+    const products = productsList.map((product) => ({
+      ...product,
+      productUrl: `mock-selfcare.base/token-exchange?institutionId=${institutionId}&productId=mock-prod-id`,
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/api/external-registries/__test__/External-registries.api.test.ts:53]: create a productsFe into mock file and moove the map code into it

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/api/external-registries/__test__/External-registries.api.test.ts (hunk 7 above)
Comment (alessandrogelmi) [reply packages/pn-pa-webapp/src/api/external-registries/__test__/External-registries.api.test.ts:53]: We left it here because we have instituionId that is not included in prodcuts. Maybe we can create a function into mock file that take InstituionId as parameter and generate the product urls. Ok?

File: packages/pn-commons/src/components/Header/__test__/Header.test.tsx [hunk 8]
```diff
@@ -61,35 +66,16 @@ describe('Header Component', () => {
     await waitFor(() => expect(handleClick).toBeCalledTimes(1));
   });
 
-  it.skip('renders header (two products, no parties and no user dropdown)', async () => {
```
[the remark this answers is not in the document]
Comment (SarahDonvito) [reply packages/pn-commons/src/components/Header/__test__/Header.test.tsx:64]: We restored it, updating it

File: packages/pn-commons/src/components/Header/__test__/Header.test.tsx (hunk 1 above)
Comment (SarahDonvito) [reply packages/pn-commons/src/components/Header/__test__/Header.test.tsx:89]: removed and it works

File: packages/pn-commons/src/components/Header/__test__/Header.test.tsx (hunk 2 above)
Comment (SarahDonvito) [reply packages/pn-commons/src/components/Header/__test__/Header.test.tsx:125]: removed and it works

File: packages/pn-pa-webapp/src/__mocks__/User.mock.ts (hunk 5 above)
Comment (SarahDonvito) [reply packages/pn-pa-webapp/src/__mocks__/User.mock.ts:3]: no, removed
